### PR TITLE
Bug 1856957 - Translations UI Download Cycle Spinner.

### PR DESCRIPTION
**Pull request by iorgamgabriel**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1856957

### CODE REVIEW

**Comment by iorgamgabriel** [comment]
https://github.com/mozilla-mobile/firefox-android/assets/8118371/d8dcbc41-c6ec-4f62-b04f-a3d966979587

File: fenix/app/src/main/res/drawable/ic_download_circle_spinner.xml
Comment (gabrielluong) [inline fenix/app/src/main/res/drawable/ic_download_circle_spinner.xml:1]: When it comes to adding icons, we should pull the resource from https://github.com/FirefoxUX/acorn-icons/tree/main/icons/mobile. You will see the icons that are available in our iconography in this [figma](https://www.figma.com/file/Kv7wWr8vH6AQrGhu9vmjdE/Mobile-Icons?node-id=221%3A1223&mode=dev). These icons will have their paths optimized and formatted in acorn-icons, which we should treat as our source of truth.

In this case, we didn't need to import this because it should be the equivalent of [mozac_ic_sync_24.xml](https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/ui/icons/src/main/res/drawable/mozac_ic_sync_24.xml). I am in the middle of updating our existing icons to use the latest paths in acorn-icons as part of [Bug 1842389](https://bugzilla.mozilla.org/show_bug.cgi?id=1842389). For the time being, I think we can remove this newly added icon and just use `mozac_ic_sync_24.xml`

File: fenix/app/src/main/java/org/mozilla/fenix/translations/LoadingIndicator.kt
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/translations/LoadingIndicator.kt:46]: Could we move the Preview to the bottom of the file?

File: fenix/app/src/main/java/org/mozilla/fenix/translations/LoadingIndicator.kt
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/translations/LoadingIndicator.kt:105]: Since our use case for the download indicator is inside of a `PrimaryButton`, I think we should be looking to use a `Button` here instead of a `Row`. Would it make sense to replace this `Row` with the `Button` definition in https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt#54-61 for the time being?

Comment (gabrielluong) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/LoadingIndicator.kt:105]: Actually, I am not fully correct here, we also have an use case with the loading indicator as a standalone list item. See https://www.figma.com/file/GT6oZLFw32LiCwKl5GOMWD/Full-Page-Translation?node-id=7341%3A209640&mode=dev 

Screen Shot 2023-10-06 at 11 53 09 AM

Comment (gabrielluong) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/LoadingIndicator.kt:105]: Here's some changes I am looking for:
- We should have a `LoadingIndicator` that displays the icon only (without the text) and handles the rotation animation as its own composable. This will be used in a `ListItem` as part of the `afterListAction` as seen in https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/compose/list/ListItem.kt#223. This will handle the use case in "Download languages" screen above and can also be reused in the Button use case.
- We should change your existing composable with the icon and text to be the Button use case. This means changing the `Row` to a `Button` as described earlier. I am specifically looking for you to split up the indicator to its own Composable that could be reuse for both use cases. I think we should make the button standalone for now instead of changing our existing [Button](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt) to accept other composable `content` for the time being and think about it as a follow up after we land our first iteration of this button.

Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/LoadingIndicator.kt:105]: Hi, I don't think it should be a button because it is not clickable and at accessibility it shouldn't say "button" . I will modify all to support the use case for download language screen .

Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/LoadingIndicator.kt:105]: From figma it should be an indicator not a button

Comment (gabrielluong) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/LoadingIndicator.kt:105]: Wouldn't the use case inside of the bottom sheet be a button? https://www.figma.com/file/GT6oZLFw32LiCwKl5GOMWD/Full-Page-Translation?node-id=7202%3A99073&mode=dev

Screen Shot 2023-10-06 at 12 20 12 PM

Comment (gabrielluong) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/LoadingIndicator.kt:105]: We're also expecting this to be a button in the bottom as described in the accessibility notes in https://www.figma.com/file/GT6oZLFw32LiCwKl5GOMWD/Full-Page-Translation?node-id=7224%3A198420&mode=dev

Screen Shot 2023-10-06 at 12 22 11 PM

Comment (gabrielluong) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/LoadingIndicator.kt:105]: To be clear, I am asking for 2 Composables:
- Button
- Indicator

Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/LoadingIndicator.kt:105]: https://www.figma.com/file/GT6oZLFw32LiCwKl5GOMWD?node-id=7221:186934&mode=design#571221606

Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/LoadingIndicator.kt:105]: I understand two elements , but I'm not sure it should be a button .

File: fenix/app/src/main/res/drawable/ic_download_circle_spinner.xml
Comment (iorgamgabriel) [reply fenix/app/src/main/res/drawable/ic_download_circle_spinner.xml:1]: Done

File: fenix/app/src/main/java/org/mozilla/fenix/translations/LoadingIndicator.kt
Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/LoadingIndicator.kt:46]: Done

File: fenix/app/src/main/java/org/mozilla/fenix/translations/LoadingIndicator.kt
Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/LoadingIndicator.kt:105]: Done

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (ohall-m) [inline fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:95]: I tested this component in TalkBack and I think it may need some accessibility refinement. Currently, TalkBack only says "Translating" on the moving spinner icon.

I looked around in our code base some more and an indeterminate progress bar has recently been added via [shopping](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProgressIndicator.kt#31,41,51).

To test the shopping loading icon feature:
1. Turn on TalkBack
2. Navigate to Walmart.com and select a more uncommon item (I picked things like grocery items)
3. Go to "Open review checker" in toolbar
4. "Check review quality" or something similar for an already processed item
5. Spinner will show up
6. Navigate to spinner
7. Talkback announces:  "In Progress", "Progress Bar". I think "In Progress" is the `<value>` and "Progress Bar" is the `<role>`. I looked in the [layout inspector semantics](https://developer.android.com/jetpack/compose/semantics) and it looks like it inherited from "ProgressBarRangeInfo". (Aside, I think it probably should have became a live region to hear that announcement without needing to manually navigate to it, but it could be my TalkBack settings?)

We used an [indeterminate progress circle indicator](https://developer.android.com/jetpack/compose/components/progress) here to make a [progress indicator](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProgressIndicator.kt#31,41,51).

A quick look at [CircularProgressIndicator](https://developer.android.com/reference/com/google/android/material/progressindicator/CircularProgressIndicator) doesn't seem to have a way to override the icon/animation (could be missing something here, please double check).

I'm thinking we will need to update this component to use [custom semantics](https://developer.android.com/jetpack/compose/accessibility) to make this have the `ProgressBarRangeInfo` [role](https://developer.android.com/reference/kotlin/androidx/compose/ui/semantics/SemanticsProperties#ProgressBarRangeInfo()).

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (ohall-m) [inline fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:132]: Just leaving a note for reference - I think `enabled = false,` is the correct state  based on this [conversation](https://www.figma.com/file/GT6oZLFw32LiCwKl5GOMWD?node-id=7221:186934&mode=design#571221606) -> "not sure it should stay a button, but if so, it should be programmatically `disabled`". 

The TalkBack response for the button seemed as expected for `DownloadIndicatorPreview` (besides the progress bar issue mentioned elsewhere).

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (ohall-m) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:95]: @anna-yeddi, does this seem to be the right direction?

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (ohall-m) [inline fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:200]: Thanks for the previews for example usage!

**Review by ohall-m — CHANGES_REQUESTED** [review]
Thanks for this patch, I tested it and it looks good! 

I think if you update the role to be `ProgressBarRangeInfo` on the spinner element, then it will provide the correct TalkBack responses.

File: fenix/app/src/main/java/org/mozilla/fenix/translations/LoadingIndicator.kt
Comment (ohall-m) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/LoadingIndicator.kt]: Thanks for the patch! I think the `Indicator` should have the role `ProgressBarRangeInfo`. I left some more info below.

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:171]: Let's remove these Base composables. Unlike our design components, I don't think we need such wrapper classes over the Material `Icon`.

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:163]:
```suggestion
            modifier = Modifier.clearAndSetSemantics {},
```
nit: Prefer `{}` over `{ }`. Same in other places.

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:79]: A bit of a naive question - is this `wrapContentSize()` doing anything for else? I have tried this with and without this and it appears the same in the preview. So, I am wondering what this is doing.

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:163]: Why is `clearAndSetSemantics` needed here? This doesn't exist in our current [Button](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt#44) implementation.

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:71]: Now that I have another chance to re-examine everything and see what parameters we need for the composable, I think I was wrong about needing to copy over over the base [Button](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt#44) in order to support our button with an animated icon and label. All we needed to add in `PrimaryButton` is an additional parameter to support `iconModifier`. 

I would propose either (1) making the changes to use PrimaryButton in this PR or (2) follow up with extending and reusing PrimaryButton.

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:44]: Done

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:47]: Done

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:171]: Done

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:163]: Done

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt]: Done I refactor it using Primary Button.

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:50]: Done

Comment (ohall-m) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:50]: Thank you for all the work on this. I think this solution will meet accessibility criteria with just a few more adjustments. 

So, this patch as it stands is manually overriding everything to provide an accurate screen reader description. However, we are losing the semantic value of the components. I think this might be okay, but needs more discussion.

So originally, I hoped we could do this:
```
        modifier = Modifier.semantics {
            role = Role.ProgressBarRangeInfo
        }
```

([Example](https://bryanherbst.com/2021/02/19/compose-role-semantics/). However, it seems compose does not support this (wasn't in the list of IDE choices when I imported role.)). If this was possible, then we would have gotten a nice inherited `role` and `value` from the Progress Bar.

I also didn't see a way to customize a [`CircularProgressIndicator()`](https://developer.android.com/reference/com/google/android/material/progressindicator/CircularProgressIndicator) for our needs e.g, changing the drawable to our sync icon. 


**I have a question:**
1. I think we should remove `clearAndSetSemantics` or could you provide further justification? That is overriding the semantic meaning of the elements and generally we need special reasoning to do so. I also do not see its use in other places in our code base. 

I'm bringing it up because setting the contentDescription this way via `clearAndSetSemantics` to "Translating in Progress" is losing the `role` of button and that it is not-enabled.

**My recommendation:**
* Please remove `clearAndSetSemantics` - it is overriding too much semantic content, unless there is further justification
* Add a parameter for us to set the `contentDescription` on the elements as a freestanding option 
* Set the default `contentDescription` of the icon to a string of `In progress`.
* Keep the default `contentDescription` of the translating button to a string of `Translating in progress`
*  (We will also need a `Downloading in progress` string in the future for use on the download screen)
* I'll add a note on the Figma about this to see if we get further recommendations and notify Emily of the string need here. (We can return and refine this later as well. I think these recommendations will push it fairly close to where it needs to be.)

Comment (ohall-m) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:50]: (Aside note, I think when we add this to the sheet proper, we will want the containing area to be a polite live-region to notify the user of the button change.)

File: fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt:56]: Let's reorder the `enabled` to be after the `modifier` parameter. We want to keep the lambda/content parameters as the last parameter so that we can `Button { onButtonClick() }` or `Button { Icon() }` if it was a `content` parameter.

File: fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt:113]:
```suggestion
    enabled: Boolean = true,
```
I think this should be understood since it's the same naming used for the Material `Button` composable.

File: fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt:125]: We want to reorder below `modifier` after making the changes above.

File: fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt:113]: Can you also add these to the other Buttons (eg Secondary, Destructive, etc)?

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:46]:
```suggestion
 * @param tint Tint [Color] to be applied to the icon.
```

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:47]:
```suggestion
 * @param iconContentDescription Optional content description for the icon.
```
Prefer plain english for the description. Code and variables should be wrapped in `[]`.

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:78]:
```suggestion
 * Download indicator for translations screens.
 * It indicates that the download of the language file is in progress.
 *
 * @param modifier [Modifier] to be applied to the layout.
 * @param text The download indicator text to be displayed.
 * @param contentDescription Content description to be applied to the button.
 * @param icon Optional [Painter] used to display an [Icon] before the download indicator text.
```

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:133]: nit should this be indented?

**Review by gabrielluong — COMMENTED** [review]
This has my conditional approval assuming my remaining feedback are approval and will not need another review from me. I will leave it to @ohall-m to provide the final approval to land. Everything is looking pretty good to me. @iorgamgabriel  Thank you for your patience in all of this!

File: fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt
Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt:113]: Done

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:46]: Done

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt]: Done

File: fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt
Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt:125]: Done

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:47]: Done

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (ohall-m) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:52]: Thanks for discussing this issue in Slack with me! 

@iorgamgabriel found a great solution. We are just waiting to get a little more information on the preferred labeling to know which route to take and this patch will be good to go with a quick update.  🎉

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:81]:
```suggestion
    text: String,
    modifier: Modifier = Modifier,
```

File: fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt
Comment (gabrielluong) [reply fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt:56]: Still looking for this change

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (ohall-m) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:52]: Have some more info on the labling:

TalkBack should report: "Translating in progress, button, disabled"

via what you shared earlier: 

```
    modifier = modifier.then(
        Modifier
            .clearAndSetSemantics {
                disabled()
                role = Role.Button
                contentDescription?.let { this.contentDescription = contentDescription }
            },
    ),
```

And for the indicator, "Download in progress" as the content description. (More context in the Figma, if needed too.)

With that change and what @gabrielluong mentioned, and we should be good to go.

File: fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt
Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt:56]: Done

File: fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt
Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt]: Done

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:81]: Done

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt:54]: Done

File: fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt:113]: I am also expecting the `iconModifier` param for the other buttons.

Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt:113]: Done

File: fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt
Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt]: Done

**Review by gabrielluong — APPROVED** [review]

**Pull request merged**
